### ISSUE TITLE
Sidebar doesn't work correctly on IE11

### ISSUE THREAD

**Issue by RojasAB** [stephanrauh/ngx-extended-pdf-viewer#123]
**Check this first**
- Does the bug also happen on https://mozilla.github.io/pdf.js/web/viewer.htm?
- Does the bug also happen on https://pdfviewer.net/simple?
only happens on the second link but for mobile version that contains a pdf with more than 1 page

**Describe the bug**
When we have a pdf with more than one page, we open the side bar
and we select any page diffrente than the current one, on IE11 we lose 
get redirected to any incorrect address instead of scroll inside the pdfviwer

**Version info**
- Version of ngx-extended-pdf-viewer
- Version of the JavaScript files pdf.js, pdf-worker.js and viewer.js (currently: default files vs. version 2.2 preview files)

any verdion that works on IE11

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: IE 11

**To Reproduce**
1. open a pdf with more than 1 page
2. click on any other page diffrent than current on the sidebar

**Comment by stephanrauh** [comment]
I'll have a look at it ASAP. I assume you've used the `*-es5` versions of the JavaScript files?
(probably yes, because AFAIK IE11 doesn't display anything with the standard files. Still, I've learned to double-check everything. Be patient with me!) :)

**Comment by stephanrauh** [comment]
I've managed to reproduce this bug. Looks like a missing polyfill. It's just hard to say which one. Stay tuned!

**Comment by RojasAB** [comment]
Thank you very much, if there is any code which i can do to support this
project please let me know

**Comment by lhomme-a** [comment]
Hi,

I think the pb is in 
src/assets/viewer.js line 2052  :  
`var pageScrollEvent = new CustomEvent('page-change');`

the  CustomEvent() constructor is not implemented in IE 
see https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent

maybe it can be replaced with

```
var pageScrollEvent = document.createEvent( 'CustomEvent' );
pageScrollEvent .initCustomEvent( 'page-change');
```

**Comment by lhomme-a** [comment]
a workaround for an angular application 
you can add this to the end of your polyfills.ts 


```
(function () {
  if (typeof (window as any).CustomEvent === "function") return false;
  function CustomEvent(event, params) {
    params = params || { bubbles: false, cancelable: false, detail: null };
    var evt = document.createEvent('CustomEvent');
    evt.initCustomEvent(event, params.bubbles, params.cancelable, params.detail);
    return evt;
  }
  (window as any).CustomEvent = CustomEvent;
})();
```

**Comment by stephanrauh** [comment]
Currently, I'm sort of stuck because I can't run my showcase with IE11, but I don't understand the error message:

![grafik](https://user-images.githubusercontent.com/3045767/64734631-49129800-d4e7-11e9-9379-523522d63f96.png)

Does this ring a bell? Can you help me? I'm afraid I can't apply your polyfill, @lhomme-a, because the application crashes earlier. However, the test application https://github.com/stephanrauh/ngx-extended-pdf-viewer does work, even with IE11. Any ideas?

**Comment by cdeiderich** [comment]
For what it's worth, I only started getting the "colon expected" error after following the instructions to move the pdf-worker.es5 from scripts to assets in angular.json, and manually setting the option in defaultOptions. I also only get the error during production builds of the project (works fine in IE 11 locally). After backing out the changes in angular.json, the colon error goes away.

**Comment by stephanrauh** [comment]
@cdeiderich I've finally found the reason of the "colon expected" error messages. My demo project set the URL of the service worker to "pdf.worker.js" without checking if it's running on IE11 or not. Such a stupid bug!

**Comment by stephanrauh** [comment]
@RojasAB @lhomme-a @CMadden8 You were right, it was "just" the missing polyfill. I've added it to the ES5 version.

**Comment by stephanrauh** [comment]
Having published version 1.7.0, I'm closing this ticket now. For some reason, `https://pdfviewer.net` still shows errors, but I guess that's because it's still loading the ES5 version of the service worker.

I'd love to hear from you. Does the new version solve the bug?

**Comment by RojasAB** [comment]
Hi stephanrauh, we try to fix it by ourself, i ignore if the latest version
fix that issue
